gensim/models/keyedvectors.py

```diff
@@ -754,8 +827,9 @@ def wmdistance(self, document1, document2):
                 if t2 not in docset2 or distance_matrix[i, j] != 0.0:
                     continue
 
-                # Compute Euclidean distance between word vectors.
-                distance_matrix[i, j] = distance_matrix[j, i] = sqrt(np_sum((self[t1] - self[t2])**2))
+                # Compute Euclidean distance between unit-normed word vectors.
```

Comment:
Why the change to L2 norm?

Reply:
Hmm, I can't recall. I generally think when computing *euclidean* distances, that usually makes more sense to do on unit-normed vectors, so that the rank-order of neighbors is the same as with the usually-preferred cosine-distance. And I vaguely recall discussion that this was the correct or better way to do WMD. But I see in the discussion on #1094 mixed guidance: a reference implementation *doesn't* use norms, but norms might work better in practice (so maybe the ref impl doesn't use norms as an oversight, or because in its design/testing it simply assumed vectors already unit-normed), and our own `wmsimilarity()` method uses norms. So maybe this should be made configurable (esp now that unit-norms don't require a giant `vectors_norm` cache), but I have no strong preference what the default should be.